examples/common/metrics_foo_library/foo_library.cc

```diff
@@ -60,7 +66,7 @@ void foo_library::observable_counter_example(const std::string &name)
   std::string counter_name                    = name + "_observable_counter";
   auto provider                               = metrics_api::Provider::GetMeterProvider();
   nostd::shared_ptr<metrics_api::Meter> meter = provider->GetMeter(name, "1.2.0");
-  meter->CreateDoubleObservableCounter(counter_name, MeasurementFetcher::Fetcher);
+  auto counter                                = meter->CreateDoubleObservableCounter(counter_name);
```

Comment:
we don't use `counter`, also `Fetcher`.

Reply:
Good catch. Have added AddCallback method to add the callback